extensions/2.0/Khronos/KHR_materials_specular/README.md

```diff
@@ -0,0 +1,100 @@
+# KHR\_materials\_specular
+
+## Khronos 3D Formats Working Group
+
+* TODO
+
+## Acknowledgments
+
+* TODO
+
+## Status
+
+Experimental
+
+## Dependencies
+
+Written against the glTF 2.0 spec.
+
+## Overview
+
+This extension adds two parameters to the metallic-roughness material: `specular` and `specularColor`.
+
+`specular` allows users to configure the strength of the specular reflection in the dielectric BRDF. A value of zero disables the specular reflection, resulting in a pure diffuse material. The metal BRDF is not affected by the parameter.
```

Comment:
I think I have trouble understanding the specular parameter, more precisely it's connection to the physicallity of the BRDF.
What will happen to energy conservation if we alter this value?
Is there any connection to a 'real world' material property?

Reply:
>  Is there any connection to a 'real world' material property?

That's a tough question... The better "real-world property" to change the amount of specular contribution is the index of refraction. However, relying only on the IOR is problematic in our simplified model:
* The IOR (#1718) is not texturable (2D textures), because we want to also use it for the volume.
* We use Schlick Fresnel with a fixed exponent of 5.
* We do not model effects like interference, absorption, bounces between layers, ...
* The material is used to fake geometry details (cavity).

Moreover, sometimes there is the need to create artificial materials:
* "Non-reflective" or diffuse-only material: `specular` allows to disable the specular reflection independent of the view direction (in theory this is also possible with IOR=1, but unfortunately not with Schlick Fresnel). If there is no specular reflection, all energy can either be reflected from the diffuse component (diffuse-only material) or transmitted into the (homogeneous) volume below for subsurface scattering (#1726).
* Non-realistic materials: cartoony shading, technical or scientific visualization, ...

Reply:
Artists require a specular factor and a specular color in order to replicate the flexibility offered by the old spec-gloss workflow in the roughness-metallic workflow.  I have tried to not offer it and got fairly negative feedback and was forced to add specular factor and specular color in.

I believe the reason for specular factor is for materials have complex microstructure that doesn't align well with the grazing angle normal of the object.  As such these materials do not have strong fresnel effects like one expects.  Thing of a fuzzy material, it's fibers do not have normals that align with the object's normal thus it does not have strong Fresnel effects at grazing angle.  These fibers are smaller than what can be represented by the normal or bump map.  Thus the main solution here is to lower the strength of the Fresnel effect.

Reply:
Ok, I think I see.

As for aproximating physical properties that we do not yet take into account (interference, absorption, bounces) - ie the `specular` parameter - This could be a good idea if we see that we are too far of to approximate them in the future. 
If not - then we should parameterize them.
However, I think we need to find a name that is more close to _what_ it actually does (transfer energy from spec to diffuse)

As for the `specular color` parameter - as it is described to me I am against it. 
It seems to be coming from the specgloss workflow, which we  are moving away from.
I would rather we keep the physicality of achromatic dielectric F0 - ie we have only one value and not RGB texture - see Disneys BRDF:
https://disney-animation.s3.amazonaws.com/library/s2012_pbs_disney_brdf_notes_v2.pdf

Reply:
> Artists require a specular factor and a specular color in order to replicate the flexibility offered by the old spec-gloss workflow in the roughness-metallic workflow. I have tried to not offer it and got fairly negative feedback and was forced to add specular factor and specular color in.
> 
> I believe the reason for specular factor is for materials that have complex microstructure that doesn't align well with the grazing angle normal of the object. As such these materials do not have strong fresnel effects like one expects. Think of a fuzzy material, it's fibers do not have surface normals that align with the object's normal thus it does not have strong Fresnel effects at grazing angle. These fibers are smaller than what can be represented by the normal or bump map. Thus the main solution here is to lower the strength of the Fresnel effect.

Consider that among dialectrics there are quite a range of F0 values. from waters 2% up to Diamonds 18%. I work with games that use a lot of gemstones and such and it was a chock and frustration to learn that the glTF exporter from Blender didn't support specular maps. Also we need to author some artistic non physically plausible materials, the lack of specular control is quite frustrating.

Reply:
> As for the `specular color` parameter - as it is described to me I am against it.
> It seems to be coming from the specgloss workflow, which we are moving away from.
> I would rather we keep the physicality of achromatic dielectric F0 - ie we have only one value and not RGB texture - see Disneys BRDF:
> https://disney-animation.s3.amazonaws.com/library/s2012_pbs_disney_brdf_notes_v2.pdf

How would you go about creating colored reflections, like oil-film on water surfaces and soap bubbles?